projects/golang/go/Makefile

```diff
@@ -69,8 +69,23 @@ prod-release: golang-build sync-artifacts-to-s3
 prod-release-images: fetch-eks-go-archive images
 
 .PHONY: fetch-eks-go-archive
-fetch-eks-go-archive:
-	curl -L -o $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz https://distro.eks.amazonaws.com/golang-$(GIT_TAG)/releases/$(BUILD_ID)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz --create-dirs
+fetch-eks-go-archive: download-go-archive validate-go-archive-checksum
+
+.PHONY: download-go-archive
+download-go-archive:
+	curl -L -o $(VERSION_DIRECTORY)/archives/linux/amd64/go1.19.5.linux-amd64.tar.gz https://distro.eks.amazonaws.com/golang-go1.19.5/releases/1/archives/linux/amd64/go1.19.5.linux-amd64.tar.gz --create-dirs
+	curl -L -o $(VERSION_DIRECTORY)/archives/linux/amd64/go1.19.5.linux-amd64.tar.gz.sha256 https://distro.eks.amazonaws.com/golang-go1.19.5/releases/1/archives/linux/amd64/go1.19.5.linux-amd64.tar.gz.sha256 --create-dirs
+
+.PHONY: validate-go-archive-checksum
+validate-go-archive-checksum:
```

Comment:
a trick we can use here to more easily support both archive and rpm checksums using the same make target is the `Make` operator `%`, to create a ["Pattern Rule"](https://www.gnu.org/software/make/manual/html_node/Pattern-Intro.html).

The `%` in target names acts like a wildcard, and you can then use the value of that wildcard inside the target.

So, for example, if we wanted to have a single `make` target `validate-go-%s-checksum`, we could call it for both `validate-go-archive-checksum` as well as `validate-go-x86-64-rpm-checksum`; and based on the value of the wildcard (the `%`) we decied what path to look in for the given files.

This way, we don't have to repeat lines 83 through 88 for each type of file we want to validate the checksum for.

For an example of what this might look like, check out [the `test-platform` target in this makefile](https://github.com/aws/eks-distro-build-tooling/blob/main/projects/golang/go/Makefile#L195). This allows us to have a single target, `test-platform-%`, which matches against any value like `test-platform-linux` or `test-platform-darwin`, without having to repeating the test code for each and every one of them